docker-compose.yml

```diff
@@ -86,3 +86,16 @@ services:
             - "/etc/passwd:/etc/passwd:ro"
         depends_on:
             - irods-catalog-provider
+
+    minio:
+        image: quay.io/minio/minio:RELEASE.2022-10-29T06-21-33Z
+        command: server --address ":19000" --console-address ":19001" /data
+        ports:
+            - "19000:19000"
+            - "19001:19001"
+        volumes:
+          - ./minio-data:/data
```

Comment:
I think this is new ground...  do we want to persist any data beyond the lifetime of these containers?   I would think not?    The demo *should* not leave anything around after `docker-compose down`?

If we choose/agree not to persist, then we don't need the minio-data/.gitignore edit either.

Reply:
Currently the rest (non-MinIO) persists after restart, which I think is great.

Yesterday the stupid VM image (pre-cooked, for Vagrant) I used decided to suspend itself to save energy, and went to sleep. I had to Reset it. If the data wasn't persistent, I would have to re-run the whole thing from scratch, create new accounts, bucket, and so on just to return to where I started.

Considering that the demo stack doesn't pre-configure everything and just let you login and start using it, I think it's valuable to have the ability to shut down and continue with testing and experimentation later (not to mention various accidental shutdowns).

So the stack creates volumes as-is, having one more (MinIO) is no harm I think.

```
$ docker volume list
DRIVER    VOLUME NAME
local     3e488c6e5617614bc3711d5885680f8a8ba73b6fbea720befc514424f802f656
local     6a25a3876e126a7ffcdfe2f15d8190eace463b8240ae2fde2bbc623b3a26f71b
local     7e054a31594a7b50763a256bc049b05c5aeac395b9d5aaaea572c8684a48489d
local     b553b1e4acdb0987ff03845f2546437f504f371b8d4cff57f0942fe7f59374c0
local     ba25659ed5dcccadc3f2f0ebd715d447630781f05ba3e7cbb59900f9e8245790
local     cda70226cd2584c895b05428a52dd9d11647b3e521db171b0f637f75e453ff8b
```

If there was a Makefile that does everything for you (all accounts, S3 bucket, resource sscript), I think it'd be OK to have completely ephemeral volumes (and run everything with `make run`, `make clean`, etc.). But I haven't seen a "stack demo" which no one wanted to have the option to shut down and restart without losing work. I think it's necessary or desired for almost all stacks that I've seen - maybe not by all users but at least by 30-50%.

Reply:
fair points.  thanks.   hadn't considered this a place yet where 'work' would be done... but it does make sense.